osu.Game.Rulesets.Mania.Tests/TestSceneHoldNoteInput.cs

```diff
@@ -188,10 +181,30 @@ public void TestPressAtStartAndBreak()
             });
 
             assertHeadJudgement(HitResult.Perfect);
-            assertTickJudgement(HitResult.LargeTickMiss);
             assertTailJudgement(HitResult.Miss);
         }
 
+        /// <summary>
+        ///     -----[           ]-----
+        ///          xox         o
+        /// </summary>
+        [Test]
+        public void TestPressAtStartThenReleaseAndImmediatelyRepress()
+        {
+            performTest(new List<ReplayFrame>
+            {
+                new ManiaReplayFrame(time_head, ManiaAction.Key1),
+                new ManiaReplayFrame(time_head + 1),
+                new ManiaReplayFrame(time_head + 2, ManiaAction.Key1),
+                new ManiaReplayFrame(time_tail),
+            });
+
+            assertHeadJudgement(HitResult.Perfect);
+            assertCombo(1);
+            assertTailJudgement(HitResult.Meh);
+            assertCombo(1);
```

Comment:
I'm not sure this test... works as possibly intended? On reading you'd think that `assert{Head,Tail}Judgement` pass immediately after the head/tail are judged. But no, judgements are collected to a list and only checked post-facto. So the `assertCombo()` checks are really checking the combo at the end of the map only. But twice.